### DISCUSSION TITLE
Data security

### DISCUSSION THREAD

**Discussion by Freyvenn** [openai/whisper#1462] [Q&A]
Hello everyone,

I have a question about Whisper, does it keep and store any data I feed it to train itself?

I think it should not as I am installing this locally, but in my case, I used google collab.

Just concerned about using it with confidential audio files as I work in law.

**Comment by jwnacnud** [comment]
If you are running it locally, you aren't sharing any data.  If you use
their API you can opt out of data sharing.

**Reply by Freyvenn** [reply]
Is there an official post from openai stating that?

**Reply by ryanheise** [reply]
> Is there an official post from openai stating that?

Not that I'm aware of, but it wouldn't really make a difference to how you can know whether it does or not. Even if OpenAI were to tell you one thing, the ultimate source of truth is the source code itself. And this is an Open Source project which means that nothing is hidden, you can see what code you are running, and you can see if it is opening any network connections to send or receive data.

In fact, if you look at the code, you can see that it *is* opening a network connection, but only to download data, not to send data. What data is it downloading? 1) a large file containing the neural net weights, and potentially 2) the audio file if you specify a URL.

I'm not saying you should believe me either, but the code is there. For example, take a look at the file [__init__.py](https://github.com/openai/whisper/blob/main/whisper/__init__.py) as it is the only file that imports the `urllib` module in order to make network connections, and it's doing so to download not upload. The other place is [audio.py](https://github.com/openai/whisper/blob/main/whisper/audio.py) which downloads the audio file via `ffmpeg`. Poke around these files, as nothing is hidden, it's all laid bare.

(By the way, even if you are not able to understand the code, as others have pointed out elsewhere, you can also unplug your Internet connection before running Whisper and then you'd be fairly certain that it didn't send any data over the Internet.)

**Comment by nicholasgcotton** [comment]
I have a guide to getting Whisper set up to run completely offline here: [https://github.com/nicholasgcotton/WhisperDO](https://github.com/nicholasgcotton/WhisperDO). IT's easiest to install it while online, but after a couple tweaks you can run Whisper without any network access at all. Why trust OpenAI when you don't have to?

**Reply by phineas-pta** [reply]
not everyone using windows

**Reply by nicholasgcotton** [reply]
I think most of the steps I outline can be made to work with Linux, you just have to figure out the .cache and vocab paths for yourself as I don't currently have a Linux box to test on.

**Comment by Freyvenn** [comment]
You're amazing, thank you for this!

________________________________
